core/domain/user_domain.py

```diff
@@ -1481,3 +1481,133 @@ def to_dict(self) -> ExplorationUserDataDict:
             'most_recently_reached_checkpoint_state_name': (
                 self.most_recently_reached_checkpoint_state_name)
         }
+
+
+class LearnerGroupUserDict(TypedDict):
+    """Dictionary for LearnerGroupUser domain object."""
+
+    user_id: str
+    invited_to_learner_groups_ids: List[str]
+    student_of_learner_groups_ids: List[str]
+    progress_sharing_permissions: List[ProgressSharingPermissionDict]
+
+
+class ProgressSharingPermissionDict(TypedDict):
+    """Dictionary for Progress Sharing Permissions of learner groups."""
+
+    group_id: str
+    sharing_is_turned_on: bool
+
+
+class ProgressSharingPermission:
+    """Domain object for Progress Sharing Permissions of learner groups."""
+
+    def __init__(
+        self,
+        group_id: str,
+        sharing_is_turned_on: bool
+    ) -> None:
+        """Constructs a ProgressSharingPermission domain object.
+
+        Attributes:
+            group_id: str. The id of the learner group.
+            sharing_is_turned_on: bool. Whether progress sharing is turned on
+                for the learner group.
+        """
+        self.group_id = group_id
+        self.sharing_is_turned_on = sharing_is_turned_on
+
+    def to_dict(self) -> ProgressSharingPermissionDict:
+        """Convert the ProgressSharingPermission domain instance into a
+        dictionary form with its keys as the attributes of this class.
+
+        Returns:
+            dict. A dictionary containing the ProgressSharingPermission class
+            information in a dictionary form.
+        """
+
+        return {
+            'group_id': self.group_id,
+            'sharing_is_turned_on': self.sharing_is_turned_on
+        }
+
+
+class LearnerGroupUser:
+    """Domain object for learner group user."""
+
+    def __init__(
+        self,
+        user_id: str,
+        invited_to_learner_groups_ids: List[str],
+        student_of_learner_groups_ids: List[str],
+        progress_sharing_permissions: List[ProgressSharingPermission]
+    ) -> None:
+        """Constructs a LearnerGroupUser domain object.
+
+        Attributes:
+            user_id: str. The user id.
+            invited_to_learner_groups_ids: list(str). List of learner group ids
+                that the user has been invited to.
+            student_of_learner_groups_ids: list(str). List of learner group ids
+                that the user is a student of.
+            progress_sharing_permissions:
+                list(ProgressSharingPermission). List of Progress Sharing
+                Permissions of learner groups.
+        """
+        self.user_id = user_id
+        self.invited_to_learner_groups_ids = invited_to_learner_groups_ids
+        self.student_of_learner_groups_ids = student_of_learner_groups_ids
+        self.progress_sharing_permissions = (
```

Comment:
```suggestion
        self.progress_sharing_permissions = progress_sharing_permissions
```
I think this will fit in a single line

Reply:
Updated.